airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -19,6 +19,36 @@
 from .utils import ShopifyRateLimiter as limiter
 
 
+class ShopifySingleRecordStream(HttpStream, ABC):
```

Comment:
I believe you don't have to implement the new main class for your `shop` stream.
You can just add the `class Shop(ShopifyStream)` and all should work under the hood. If you have any issues with this, please provide the proper comments, we will try to figure it out together.

Reply:
I have changed it to inherit, but I had to override the `parse_response` method because `ShopifyStream` is expecting an array of records in the response, where the `shop` endpoint's response is just a single object.

Reply:
You can redefine the `parse_response` method inside of `Shop(ShopifyStream)` and adapt it as expected.

Reply:
that is what I have done

Reply:
https://github.com/airbytehq/airbyte/pull/8297/files#diff-e437ad3d7dcb3c22d1c955775eaa362262214ed513264b7fa1c9521e7658902aR401